// @opentelemetry/exporter-prometheus/build/src/PrometheusLabelsBatcher.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrometheusLabelsBatcher = void 0;
class PrometheusLabelsBatcher {
    constructor() {
        this._batchMap = new Map();
    }
    get hasMetric() {
        return this._batchMap.size > 0;
    }
    process(record) {
        const name = record.descriptor.name;
        let item = this._batchMap.get(name);
        if (item === undefined) {
            item = {
                descriptor: record.descriptor,
                aggregatorKind: record.aggregator.kind,
                records: new Map(),
            };
            this._batchMap.set(name, item);
        }
        const recordMap = item.records;
        const labels = Object.keys(record.labels)
            .map(k => `${k}=${record.labels[k]}`)
            .join(',');
        recordMap.set(labels, record);
    }
    checkPointSet() {
        return Array.from(this._batchMap.values()).map(({ descriptor, aggregatorKind, records }) => {
            return {
                descriptor,
                aggregatorKind,
                records: Array.from(records.values()),
            };
        });
    }
}
exports.PrometheusLabelsBatcher = PrometheusLabelsBatcher;
